CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+# Development guide
+
+## Compiling / testing via the command-line tool
+
+The [command-line tool](https://github.com/alexarchambault/coursier#command-line-1) of coursier allows to quickly test changes in the `core`, `cache`, and `cli` modules. To get a freshly compiled launcher for it, type
+```
+$ sbt ++2.11.8 cli/pack
+```
+
+`++2.11.8` sets the Scala version to `2.11.8`. This is the latest `2.11` Scala version. The launcher is only compiled for Scala `2.11` for now.
+
+`cli/pack` instructs sbt to create a "pack" directory, via the [sbt-pack](https://github.com/xerial/sbt-pack) plugin. Its result can be found under `cli/target/pack`. It mainly contains a `lib` sub-directory, containing the full launcher classpath, along with a `bin` sub-directory, that contains shell scripts allowing to launch applications contained in `lib`.
+
+As the launcher has only one main class, `cli/target/pack/bin` contains only one launcher, named `coursier`. It can be launched like
+```
+$ cli/target/pack/bin/coursier --help
+```
+
+During development, it can be helpful to have the "pack" directory mirror what's in the sources. This can be achieved with
+```
+$ sbt ++2.11.8 "~cli/pack"
+```
+which instructs sbt to watch the sources of coursier, and launch `cli/pack` whenever a change is detected.
+
+Note that `cli/pack` triggers a compilation of the `cli` module, along with the modules it depends on (currently, `coreJVM`, the JVM counterpart of the cross-compiled `core` module, and `cache`). That allows to catch compilation errors along the way while generating the "pack" directory.
+
+
+## Compiling / testing the sbt plugin(s)
+
+### Via scripted
+
+[scripted](http://eed3si9n.com/testing-sbt-plugins) is a test framework for sbt plugins. `sbt-coursier` and `sbt-shading` are tested via scripted.
```

Comment:
http://www.scala-sbt.org/0.13/docs/Testing-sbt-plugins.html is the canonical URL nowadays